.github/workflows/code-quality.yml

```
@@ -0,0 +1,294 @@
+
+      - name: Run TypeScript compiler
+        run: npx tsc --noEmit --pretty
+
+      - name: Check code complexity
+        run: |
+          echo "Checking code complexity..."
+          npx complexity-report src/ --format json --output complexity-report.json || echo "Complexity report generated"
+        continue-on-error: true
+
+      - name: Upload complexity report
+        uses: actions/upload-artifact@v3
```

actionlint (1.7.7)

the runner of "actions/upload-artifact@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)